tic-tac-toe/src/components/game/game.js

```diff
@@ -0,0 +1,124 @@
+import React, {Component} from 'react';
+import Board from '../board/board';
+import ControlPannel from '../control-pannel/control-pannel';
+import GameInfo from '../info-bar/info-bar';
+import InfoBlock from '../info-block/info-block';
+import checkTurn from '../../scripts/calculate-next-move';
+import calculateWinner from '../../scripts/calculate-winner';
+import {getGameData, updateGameData} from '../../scripts/storage-controller';
+import './game.css';
+
+class Game extends Component {
+    constructor(props) {
+		super(props);
+    this.isBlockGame = false;
+		this.state = getGameData() || {
+			history: [
+				{
+					squares: Array(9).fill(null),
+				}
+			],
+			stepNumber: 0,
+			xIsNext: true,
+      activeGameMode: 'human',
+		}
+    console.log(getGameData())
+	}
+
+	handleClick(i) {
+    if (this.isBlockGame) {return}
+		const history = this.state.history.slice(0, this.state.stepNumber + 1);
+		const current = history[history.length - 1];
+		const squares = current.squares.slice();
+		if (calculateWinner(squares) || squares[i]) {
+				return;
+		}
+		squares[i] = this.state.xIsNext ? 'X' : 'O';
+		this.setState({
+				history: history.concat([{
+					squares: squares,
+				}]),
+				stepNumber: history.length,
+				xIsNext: !this.state.xIsNext,
+		});
+	}
+
+	jumpTo(step) {
+		this.setState({
+			stepNumber: step,
+			xIsNext: (step % 2) === 0,
+		});
+	}
+
+  onChangeMode(mode) {
+    this.setState({
+      activeGameMode: mode
+    })
+  }
```

Comment:
Reset your current game after changing mode.

Reply:
I made this.
```suggestion
  onChangeMode(mode) {
    this.onResetGame();
    this.setState({
      activeGameMode: mode,
    })
  }
```